woocommerce/woocommerce-ios
# Update FooterSpinnerView to have single appearance in all tables

Fixes https://github.com/woocommerce/woocommerce-ios/issues/1922.

##  Description

Initial issue - padding of the bottom loader in the Orders table view is wrong.

After some investigation in https://github.com/woocommerce/woocommerce-ios/issues/1922#issuecomment-734796730 I decided to simplify `FooterSpinnerView` layout to have single appearance independent of tableview style.

Here is old layout logic as I understand:
- since grouped tableview has excessive padding before footer view - pin activity indicator to the top of the footer, so it will appear centered (vertically)
- plain tableview displays footer view just fine, without extra padding - pin activity indicator to the center of the footer

I suggest to deal with grouped tableview padding by setting `tableView.sectionFooterHeight = .leastNonzeroMagnitude` where needed. And with that every footer spinner will appear in same way and same height in all tables.

While testing I observe that plain tableviews look just as before and grouped have slightly decreased footer size (because it had additional padding previously). For all grouped (and therefore changed) cases comparison screenshots provided below.

## Screens affected

Plain (no change):
- ProductsViewController
- SearchViewController
- ProductVariationsViewController

Grouped (footer height decreased to 45):
- OrderListViewController and OrdersViewController (already have `.leastNonzeroMagnitude` footer height and that's why indicator appeared wrong)
- ReviewsViewController
- ProductReviewsViewController
- ShipmentProvidersViewController

PaginatedListSelectorViewController plain implementations:
- LinkedProductsListSelectorViewController
- ProductListSelectorViewController

PaginatedListSelectorViewController grouped implementations:
- ProductShippingSettingsViewController
- ProductPriceSettingsViewController

## Testing

1. Open any of screens above and scroll to the bottom.
2. Observe that spinner is centered vertically.

## Screenshots

### OrderListViewController
 Before | After 
--------|------
![1](https://user-images.githubusercontent.com/3132438/100666426-09dd9a80-336a-11eb-8cef-ca80c216a9df.png)|![2](https://user-images.githubusercontent.com/3132438/100666444-106c1200-336a-11eb-8527-807e2f245141.png)

### ReviewsViewController
 Before | After 
--------|------
![1](https://user-images.githubusercontent.com/3132438/100669007-cdac3900-336d-11eb-834d-d3c4397cd0b9.png)|![2](https://user-images.githubusercontent.com/3132438/100669019-d1d85680-336d-11eb-9960-f94e44c5e76f.png)

### ProductReviewsViewController
 Before | After 
--------|------
![1](https://user-images.githubusercontent.com/3132438/100669107-f2081580-336d-11eb-903f-3de26918ee07.png)|![2](https://user-images.githubusercontent.com/3132438/100669114-f6ccc980-336d-11eb-8166-f6c8321d3905.png)

### ShipmentProvidersViewController
 Before | After 
--------|------
![1](https://user-images.githubusercontent.com/3132438/100669133-021ff500-336e-11eb-95d4-9f7fc88a3578.png)|![2](https://user-images.githubusercontent.com/3132438/100669148-064c1280-336e-11eb-909d-bc6a1ccc79ec.png)

### ProductShippingSettingsViewController
 Before | After 
--------|------
![Shipping1](https://user-images.githubusercontent.com/3132438/100669175-1106a780-336e-11eb-87be-4d320263ac63.png)|![Shipping2](https://user-images.githubusercontent.com/3132438/100669187-1663f200-336e-11eb-81f8-97c6da79c3ae.png)

### ProductPriceSettingsViewController
 Before | After 
--------|------
![Price1](https://user-images.githubusercontent.com/3132438/100669218-2085f080-336e-11eb-8516-beaed9b15c73.png)|![Price2](https://user-images.githubusercontent.com/3132438/100669228-24197780-336e-11eb-8c66-08651936f9bd.png)


Update release notes:
